app/liquid_tags/unified_embed/tag.rb

```diff
@@ -24,31 +25,28 @@ class Tag < LiquidTagBase
     def self.new(tag_name, input, parse_context)
       stripped_input = ActionController::Base.helpers.strip_tags(input).strip
 
-      # Extract just the URL from the input, without any params, for validation
-      actual_link = extract_only_url(stripped_input)
-
-      # When Listings are disabled, it makes little sense to perform a validate_link
+      # when Listings are disabled, it makes little sense to perform a validate_link
       # network call.
-      handle_listings_disabled!(actual_link)
+      handle_listings_disabled!(stripped_input)
 
       # Before matching against the embed registry, we check if the link
       # is valid (e.g. no typos).
       # If the link is invalid, we raise an error encouraging the user to
       # check their link and try again.
-      validate_link!(actual_link)
-      klass = UnifiedEmbed::Registry.find_liquid_tag_for(link: stripped_input)
+      validated_link = validate_link(stripped_input)
+      klass = UnifiedEmbed::Registry.find_liquid_tag_for(link: validated_link)
 
-      # If there are no OG metatags, we shall render an A-tag. Since the link
-      # has been validated, at least this A-tag will not 404.
-      #
       # Why the __send__?  Because a LiquidTagBase class "privatizes"
       # the `.new` method.  And we want to instantiate the specific
       # liquid tag for the given link.
-      klass.__send__(:new, tag_name, stripped_input, parse_context)
+      klass.__send__(:new, tag_name, validated_link, parse_context)
     end
 
-    def self.validate_link!(link)
-      uri = URI.parse(link)
+    def self.validate_link(link, limit = MAX_REDIRECTION_COUNT)
+      # Extract just the URL from the input, without any params, for validation
+      actual_link = extract_only_url(link)
```

Comment:
Let me explain why I moved the `actual_link` variable in here, and refactored the `validate_link` method to return a link.
In the case of a redirect, it doesn't make sense to proceed with the original embedded link; we need to proceed with the validated redirected_to link.
`actual_link` is the embedded link without any params (which consistently failed validation when I previously allowed them). So we use `actual_link` to validate, but then proceed either with the embedded link, OR with the rediected_to link.

Please let me know if this doesn't make sense. 🙏🏾

Reply:
makes sense to me! small nit -- can we change `limit` to `retries` or something similar? I feel like it reads a tad better. especially as `retries.zero?`